### PR TITLE
Add #error to Spanner Instance::Job and Database::Job

### PR DESCRIPTION
**Pull request by quartzmo**
Also, raise a meaningful error if a job errors in Spanner acceptance setup.

### CODE REVIEW

File: google-cloud-spanner/lib/google/cloud/spanner/database/job.rb [hunk 1]
```diff
@@ -122,6 +122,32 @@ def error?
...
+          #   job = spanner.create_database "my-instance",
+          #                                 "my-new-database"
+          #
+          #   job.error? # true
+          #
+          #   error = job.error
+          #
+          def error
+            return nil unless error?
+            fail GRPC::BadStatus.new @grpc.error.code, @grpc.error.message
```
Comment (blowmage) [inline google-cloud-spanner/lib/google/cloud/spanner/database/job.rb]: In order to set `Error#cause` we need to raise the error. So I suppose this code should look more like this:

```ruby
begin
  begin
    fail GRPC::BadStatus.new @grpc.error.code, @grpc.error.message
  rescue GRPC::BadStatus => grpc_error
    fail Google::Cloud::Error.from_error grpc_error
  end
rescue Google::Cloud::Error => error
  return error
end
```

Or, we could just return the Google::Rpc::Status object, leaking the low level API.

Comment (quartzmo) [inline google-cloud-spanner/lib/google/cloud/spanner/database/job.rb]: While I'm not in favor of leaking the low-level API, I am beginning to think a status object is more appropriate than returning an un-raised Error, which just seems a little off-pattern.

Comment (quartzmo) [inline google-cloud-spanner/lib/google/cloud/spanner/database/job.rb]: I'll update this PR to use a status object of some sort.

File: google-cloud-spanner/lib/google/cloud/spanner/status.rb [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        def initialize code, message, details
+          @code = code
+          @message = message
+          @details = details
+        end
+
+        ##
+        # @private New Status from a Google::Rpc::Status object.
+        def self.from_grpc grpc
+          code_sym = grpc_code_description_for grpc.code
```
Comment (blowmage) [inline google-cloud-spanner/lib/google/cloud/spanner/status.rb]: I would rather see the code symbol be a separate attribute. The numeric code value will be valuable.

**Pull request merged**